dio/test/stacktrace_test.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:dio/dio.dart';
+import 'package:stack_trace/stack_trace.dart';
+import 'package:test/test.dart';
+
+import 'mock/adapters.dart';
+
+void main() async {
+  group('useStackTraceChains', () {
+    test('enabled', () async {
+      final dio = Dio(BaseOptions(useStackTraceChains: true))
+        ..httpClientAdapter = MockAdapter()
+        ..options.baseUrl = MockAdapter.mockBase;
+
+      try {
+        await dio.get('/foo');
+      } catch (e, stackTrace) {
+        expect(stackTrace, isA<Chain>());
+        expect(stackTrace.toString(), contains('test/stacktrace_test.dart'));
```

Comment:
We have there correct stacktrace here, something like this:

```
package:dio/src/dio_mixin.dart 556:7               DioMixin.fetch.<fn>
===== asynchronous gap ===========================
dart:async                                         Future.catchError
package:dio/src/dio_mixin.dart 549:8               DioMixin.fetch
package:dio/src/dio_mixin.dart 392:36              DioMixin.request.<fn>.<fn>
===== asynchronous gap ===========================
dart:async                                         new Future
package:dio/src/dio_mixin.dart 365:15              DioMixin.request.<fn>
package:stack_trace                                Chain.capture
package:dio/src/dio_mixin.dart 364:13              DioMixin.request
package:dio/src/dio_mixin.dart 63:12               DioMixin.get
test/stacktrace_test.dart 15:19                    main.<fn>.<fn>
package:test_api/src/backend/declarer.dart 215:19  Declarer.test.<fn>.<fn>

```

Reply:
This looks pretty good, but I am not sure if tooling like Sentry/Crashlytics can work with this.